environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaCreationRetrievalTask.java

```diff
@@ -19,44 +22,52 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(FreeIpaCreationRetrievalTask.class);
 
+    private final FreeIpaService freeIpaService;
+
+    public FreeIpaCreationRetrievalTask(FreeIpaService freeIpaService) {
+        this.freeIpaService = freeIpaService;
+    }
+
     @Override
     public boolean checkStatus(FreeIpaPollerObject freeIpaPollerObject) {
         String environmentCrn = freeIpaPollerObject.getEnvironmentCrn();
         try {
             LOGGER.info("Checking the state of FreeIpa creation progress for environment: '{}'", environmentCrn);
-            DescribeFreeIpaResponse freeIpa = freeIpaPollerObject.getFreeIpaV1Endpoint().describe(environmentCrn);
-            if (freeIpa == null) {
-                throw new FreeIpaOperationFailedException("FreeIpa instance does not exist.");
-            } else if (freeIpa.getStatus().isDeletionInProgress() || freeIpa.getStatus().isSuccessfullyDeleted()) {
-                LOGGER.info("FreeIpa '{}' '{}' is getting terminated (status:'{}'), polling is cancelled.",
-                        freeIpa.getName(),
-                        freeIpa.getCrn(),
-                        freeIpa.getStatus());
+            Optional<DescribeFreeIpaResponse> freeIpa = freeIpaService.describe(environmentCrn);
+            if (freeIpa.isEmpty()) {
+                throw new FreeIpaOperationFailedException("FreeIpa cluster not found for environment: " + environmentCrn);
+            } else if (freeIpa.get().getStatus().isDeletionInProgress() || freeIpa.get().getStatus().isSuccessfullyDeleted()) {
```

Comment:
hmm.
the `else` does not have any use here as `throw` will exit method anyway.
you can get the optional value and avoid using the `get()` multiple times.